test/adaptive_load/fake_plugins/fake_metrics_plugin/fake_metrics_plugin.cc

```diff
@@ -0,0 +1,67 @@
+#include "test/adaptive_load/fake_plugins/fake_metrics_plugin/fake_metrics_plugin.h"
+#include "api/adaptive_load/benchmark_result.pb.h"
+#include "envoy/common/exception.h"
+
+namespace Nighthawk {
+
+FakeMetricsPlugin::FakeMetricsPlugin(
+    const nighthawk::adaptive_load::FakeMetricsPluginConfig& config)
+    : fixed_metric_value_{config.fixed_metric_value()} {}
+
+absl::StatusOr<double> FakeMetricsPlugin::GetMetricByName(absl::string_view metric_name) {
+  if (metric_name == "bad-metric") {
+    return absl::InternalError("bad-metric requested");
+  }
+  return fixed_metric_value_;
+}
+
+const std::vector<std::string> FakeMetricsPlugin::GetAllSupportedMetricNames() const {
+  return {"good-metric", "bad-metric"};
+}
+
+std::string FakeMetricsPluginConfigFactory::name() const { return "nighthawk.fake-metrics-plugin"; }
+
+Envoy::ProtobufTypes::MessagePtr FakeMetricsPluginConfigFactory::createEmptyConfigProto() {
+  return std::make_unique<nighthawk::adaptive_load::FakeMetricsPluginConfig>();
+}
+
+MetricsPluginPtr
+FakeMetricsPluginConfigFactory::createMetricsPlugin(const Envoy::Protobuf::Message& message) {
+  const auto& any = dynamic_cast<const Envoy::ProtobufWkt::Any&>(message);
+  nighthawk::adaptive_load::FakeMetricsPluginConfig config;
+  Envoy::MessageUtil::unpackTo(any, config);
+  return std::make_unique<FakeMetricsPlugin>(config);
+}
+
+absl::Status
+FakeMetricsPluginConfigFactory::ValidateConfig(const Envoy::Protobuf::Message& message) const {
+  try {
+    const auto& any = dynamic_cast<const Envoy::ProtobufWkt::Any&>(message);
+    nighthawk::adaptive_load::FakeMetricsPluginConfig config;
+    Envoy::MessageUtil::unpackTo(any, config);
+    if (config.fixed_metric_value() < 0) {
+      return absl::InvalidArgumentError(
+          "Negative fixed_metric_value triggered validation failure.");
+    }
+    return absl::OkStatus();
+  } catch (const Envoy::EnvoyException& e) {
+    return absl::InvalidArgumentError(
+        absl::StrCat("Failed to parse FakeMetricsPluginConfig proto: ", e.what()));
+  }
+}
+  
+REGISTER_FACTORY(FakeMetricsPluginConfigFactory, MetricsPluginConfigFactory);
+
+envoy::config::core::v3::TypedExtensionConfig
+MakeFakeMetricsPluginConfig(double fixed_metric_value) {
```

Comment:
I found no existing Envoy version. I tried to create my own just now, but the trouble is that the double value is part of the hassle of setting it up.

We could get halfway there with:

template<typename T>
TypedExtensionConfig MakePluginConfig(absl::string_view plugin_name, const T& custom_config_proto);

with the caller still having to set the contents of the proto.

Reply:
I see, that makes sense. Do you think that would still be worth doing? Also, maybe we could accept a delegate arg to do the value setting for us?

Reply:
Actually deleting this particular function because we decided to make the config proto more complicated :)

But I started to try to write such a utility and found it's a bit complicated. Opened issue #468 to discuss how to do it.

Reply:
Thanks for looking into this and filing the issue, resolving this here.